File: PhoneBook/Models/PhoneBookEntry.cs
using System;
using System.Collections.Generic;
using System.Text.Json.Serialization;

namespace PhoneBook.Models;

public partial class PhoneBookEntry
{
    public long PhoneBookEntryId { get; set; }

    [JsonPropertyName("firstName")]
    public string Firstname { get; set; } = null!;

    public string Surname { get; set; } = null!;

    public string PhoneNumber { get; set; } = null!;
}

File: PhoneBook/ClientApp/src/app/models/phone-book-entry.ts
export class PhoneBookEntry {
  public phoneBookEntryId?: number;
  public firstName: string;
  public surname: string;
  public phoneNumber: string;

  constructor(firstName: string, surname: string, phoneNumber: string){
    this.firstName = firstName;
    this.surname = surname;
    this.phoneNumber = phoneNumber;
  }
}
